src/gt4py/next/iterator/embedded.py

```diff
@@ -1156,6 +1156,7 @@ def ndarray(self) -> core_defs.NDArrayObject:
     def asnumpy(self) -> np.ndarray:
         raise NotImplementedError()
 
+    # TODO(tehrengruber): Use a regular zero dimensional field instead.
```

Comment:
We moved the comment to a better place. With respect to the comment itself: The `IndexField` class is rather strange. It has two modes of operation: Either it is a field with (conceptually) `field(domain) == domain` or it is a zero-dimensional field. Both modes don't share any implementation similarities, but are mushed into the same class. The way the class behaves is then controlled using `_cur_index`. It would be much simpler to just make `field[index]` return a zero-dimensional field which is exactly what we want instead of re-implementing it here.

Reply:
I don't understand what you are saying...